Mathlib/Algebra/MonoidAlgebra/Degree.lean

```diff
@@ -232,13 +233,15 @@ theorem supDegree_sub_le {f g : R'[A]} :
   rw [sub_eq_add_neg, ← supDegree_neg (f := g)]; apply supDegree_add_le
 
 theorem supDegree_sum_le {ι} {s : Finset ι} {f : ι → R[A]} :
-    (∑ i ∈ s, f i).supDegree D ≤ s.sup (fun i => (f i).supDegree D) :=
-  (Finset.sup_mono Finsupp.support_finset_sum).trans_eq (Finset.sup_biUnion _ _)
+    (∑ i ∈ s, f i).supDegree D ≤ s.sup (fun i => (f i).supDegree D) := by
+  classical
+  exact (Finset.sup_mono Finsupp.support_finset_sum).trans_eq (Finset.sup_biUnion _ _)
 
 theorem supDegree_single_ne_zero (a : A) {r : R} (hr : r ≠ 0) :
     (single a r).supDegree D = D a := by
   rw [supDegree, Finsupp.support_single_ne_zero a hr, Finset.sup_singleton]
 
+open Classical in
```

Comment:
Is this needed, or can you add suitable `DecidableEq` hypotheses?

Reply:
I suspect the answer is "let's scoped all the `Classical`s first, and remove them later"

Reply:
For each statement, I have tried to balance between ease of use and adding decidability assumptions. If we notice that some of my choices create friction elsewhere, we can go back and fix them later.